app/src/main/java/com/feedhenry/securenativeandroidtemplate/mvp/components/AuthHelper.java

```diff
@@ -0,0 +1,162 @@
+package com.feedhenry.securenativeandroidtemplate.mvp.components;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Base64;
+import android.util.Log;
+import net.openid.appauth.AuthState;
+import org.json.JSONException;
+
+import java.io.UnsupportedEncodingException;
+import java.util.concurrent.locks.ReentrantLock;
+
+import javax.inject.Inject;
+
+import okhttp3.Call;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+
+/**
+ * Created by tjackman on 9/16/17.
+ */
+
+public class AuthHelper {
+
+    private final SharedPreferences mPrefs;
+    private final ReentrantLock mPrefsLock;
```

Comment:
is the lock really necessary? Will there be multiple threads trying to read/write the authState at the same time?